WooCommerce/Classes/Authentication/Jetpack Setup/Native Jetpack Setup/LoginJetpackSetupView.swift

```diff
@@ -3,9 +3,30 @@ import SwiftUI
 /// Hosting controller for `LoginJetpackSetupView`.
 ///
 final class LoginJetpackSetupHostingController: UIHostingController<LoginJetpackSetupView> {
-    init(siteURL: String, connectionOnly: Bool) {
-        let viewModel = LoginJetpackSetupViewModel(siteURL: siteURL, connectionOnly: connectionOnly)
+    init(siteURL: String, connectionOnly: Bool, onStoreNavigation: @escaping () -> Void) {
+        let viewModel = LoginJetpackSetupViewModel(siteURL: siteURL, connectionOnly: connectionOnly, onStoreNavigation: onStoreNavigation)
         super.init(rootView: LoginJetpackSetupView(viewModel: viewModel))
+        rootView.webViewPresentationHandler = { [weak self] in
+            guard let self else { return }
+            guard let connectionURL = viewModel.jetpackConnectionURL else {
+                return
+            }
+
+            let webViewModel = JetpackConnectionWebViewModel(initialURL: connectionURL,
+                                                             siteURL: viewModel.siteURL,
+                                                             title: Localization.approveConnection) { [weak self] in
+                viewModel.shouldPresentWebView = false
+                viewModel.authorizeJetpackConnection()
+                self?.dismissView()
+            }
+            let webView = AuthenticatedWebViewController(viewModel: webViewModel)
+            webView.navigationItem.leftBarButtonItem = UIBarButtonItem(title: Localization.cancel,
+                                                                       style: .plain,
+                                                                       target: self,
+                                                                       action: #selector(self.dismissView))
+            let navigationController = UINavigationController(rootViewController: webView)
+            self.present(navigationController, animated: true)
+        }
```

Comment:
Nit: Would it make sense to move this to a private method and name it to explain the purpose? Something like `presentWebViewToAuthorizeJetpackConnection`.